### ISSUE TITLE
Control/Read Fan Speed

### ISSUE THREAD

**Issue by dernerv** [CorsairOfficial/cue-sdk#13]
The Commander Pro supports Fan Control via iCUE but  the SDK not. 
Is it possible to add support to read/write the Fan speed.
Same for the temperature sensors.

**Comment by AdamLeyshon** [comment]
I too would also like to see fan/temperature readouts and fan control via the SDK.
I use 3rd party monitoring tool (Argus Monitor) for fan/pump control, but requires me to stop the Corsair services and iCUE so my lighting stops working.

Exposing this functionality would allow us to have full control over Corsair fan/AIO products, whilst allowing us to use RGB on all our products together.

**Comment by AdamLeyshon** [comment]
Can we update the title to also include pump speed? Read/Write Fan and Pump speeds?

**Comment by DmitrySychev** [comment]
Agreed. I'm just as interested in being able to access all of the sensors icue monitors as the fancy leds.

**Comment by Tenn0** [comment]
Also having access to the coolant temperature from my AIO would be handy. Im logging my temperatures via openhardwaremonitor into home-assistant, having also the coolant temperature would be nice to see and visualize the relation between core temp, room temp and coolant temp.

**Comment by Garulf** [comment]
I'm currently using a python script to scrape iCUE's sensor logs to achieve this. But it feels very hacky. How is this STILL not exposed by some API?

**Comment by Tenn0** [comment]
> 
> 
> I'm currently using a python script to scrape and read iCUE's sensor logs to read fan speed/coolant temps. But it feels very hacky, and there is no fan/pump control. How is this STILL not exposed by some API?

Can you pls share your script? im interested in it and wanna make use of it myself.

**Comment by Garulf** [comment]
> > I'm currently using a python script to scrape and read iCUE's sensor logs to read fan speed/coolant temps. But it feels very hacky, and there is no fan/pump control. How is this STILL not exposed by some API?
> 
> Can you pls share your script? im interested in it and wanna make use of it myself.

Yeah, I use have telegraf running the script to scrape the log file and post to my influxdb server.

https://gist.github.com/Garulf/295d6be11df7b023fc25e5198324b3ac

You may have to adjust it to suit your needs
